### PR TITLE
 Reflect v0.11.0 changes in StateUpdate

### PR DESCRIPTION
**Pull request by omerfirmak**

### CODE REVIEW

**Comment by omerfirmak** [comment]
Closes #669

File: clients/feeder/feeder_test.go [hunk 1]
```diff
@@ -374,3 +374,26 @@ func TestBackoffFailure(t *testing.T) {
...
+	t.Run("declared Cairo0 classes", func(t *testing.T) {
+		update, err := client.StateUpdate(context.Background(), 283746)
+		require.NoError(t, err)
+		assert.Equal(t, true, len(update.StateDiff.OldDeclaredContracts) > 0)
+	})
+
+	t.Run("declared Cairo1 classes", func(t *testing.T) {
+		update, err := client.StateUpdate(context.Background(), 283364)
+		require.NoError(t, err)
+		assert.Equal(t, true, len(update.StateDiff.DeclaredClasses) > 0)
```
[the remark this answers is not in the document]
Comment (omerfirmak) [reply clients/feeder/feeder_test.go:385]: Nice find, thanks! Done.

**Review by wojciechos — APPROVED** [review]

**Comment by omerfirmak** [comment]
Closes #686

**Comment by wojciechos** [comment]
I ran the PR version on mainnet, and upon reaching the Starknet block with the new version, it displayed "unsupported block version" warnings and caused 100% CPU usage. I guess we should:

- Update `supportedStarknetVersion` in `blockchain.go`
- Implement safeguards to prevent 100% CPU in case of an unsupported version

image

image

File: db/buckets.go [hunk 2]
```diff
@@ -22,6 +22,8 @@ const (
 	TransactionsByBlockNumberAndIndex       // maps block number and index to transaction
 	ReceiptsByBlockNumberAndIndex           // maps block number and index to transaction receipt
 	StateUpdatesByBlockNumber
+	ClassesRootKey
+	ClassesTrie
```
Comment (IronGauntlets) [inline db/buckets.go:26]: I don't see why we need both of these. I believe `ClassesTrie` without any suffix is sufficient to store `ClassesRootKey`. The same applies to StateRootKey and StateTrie.

File: sync/sync.go [hunk 3]
```diff
@@ -61,7 +61,7 @@ func (s *Synchronizer) fetcherTask(ctx context.Context, height uint64, verifiers
 			for _, deployedContract := range stateUpdate.StateDiff.DeployedContracts {
 				referencedClasses[*deployedContract.ClassHash] = nil
 			}
-			for _, classHash := range stateUpdate.StateDiff.DeclaredClasses {
+			for _, classHash := range stateUpdate.StateDiff.DeclaredV0Classes { // todo: fetch v1 classes as well
```
Comment (IronGauntlets) [inline sync/sync.go:64]: Please add an issue for this if not already done so

**Review by IronGauntlets — APPROVED** [review]
I made some changes [here](https://github.com/NethermindEth/juno/compare/c1ff88da5fb79acfe36b691a71b11500bdbf76f8..f73e32f8ba107ce143f79ab72091ac9f5e6c5132) and squash one of the commits.

[Rebased](https://github.com/NethermindEth/juno/compare/f73e32f8ba107ce143f79ab72091ac9f5e6c5132..3266914b94727a6912fb5227e7a8d29af649a062)

File: db/buckets.go (hunk 2 above)
Comment (omerfirmak) [reply db/buckets.go:26]: Yeap, done

**Pull request merged**